### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by antondalgren**
tsdoc is better at handling typescript and understand the types by using the typescript compiler.
In order to get "better" docs, the default exporting had to be done on a seperate line.

Based on PR #1 and commit d9cf64c

### CODE REVIEW

**Comment by antondalgren** [comment]
Would prefer if we could have the docs generated in CI instead of also commit that. Need to dig further on this.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -24,4 +24,33 @@ jobs:
...
+    - name: Download docs
+      uses: actions/download-artifact@v2
+      with:
+        name: docs
+    - name: Deploy docs with gh-pages
+      run: |
+        git remote set-url origin https://git:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+        npx gh-pages -d docs -u "github-actions-bot <support+actions@github.com>"
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (antondalgren) [reply .github/workflows/ci.yaml:56]: Yeah, could also have the gh-pages package as dev dependency and do the following in the build step.
```yml
- name: Deploy docs with gh-pages 
  run: |        
     git remote set-url origin https://git:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
     npx gh-pages -d docs -u "github-actions-bot &lt;support+actions@github.com&gt;"
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by dentarg** [comment]
Would be nice to do the same in our repos. We use the one I linked in at least one other repo :)

**Comment by antondalgren** [comment]
> Would be nice to do the same in our repos. We use the one I linked in at least one other repo :)

I agree, I was unaware that we used that action.

**Comment by antondalgren** [comment]
> Would be nice to do the same in our repos. We use the one I linked in at least one other repo :)

@dentarg Updated it to use the action you linked! :)

**Review by dentarg — APPROVED** [review]
Nice work!

Can't say that much about JS/TS things but generating docs locally worked without problems for me. :)

**Comment by dentarg** [comment]
> Can't say that much about JS/TS things

I guess the majority of that is discussed in #1 anyway

**Comment by antondalgren** [comment]
> Nice work!
> 
> Can't say that much about JS/TS things but generating docs locally worked without problems for me. :)

Yeah, those are covered in #1.

**Comment by carlhoerberg** [comment]
Fixed in: https://github.com/cloudamqp/amqp-client.js/commit/b47f3ed99d694f92f01f59db7cf053cca34095ea

**Pull request closed without merging**